opnode/test/system_test.go

```diff
@@ -641,3 +645,196 @@ func TestL1InfoContract(t *testing.T) {
 	checkInfoList("On verifier with state", l1InfosFromVerifierState)
 
 }
+
+// calcGasFees determines the actual cost of the transaction given a specific basefee
+func calcGasFees(gasUsed uint64, gasTipCap *big.Int, gasFeeCap *big.Int, baseFee *big.Int) *big.Int {
+	x := new(big.Int).Add(gasTipCap, baseFee)
+	// If tip + basefee > gas fee cap, clamp it to the gas fee cap
+	if x.Cmp(gasFeeCap) > 0 {
+		x = gasFeeCap
+	}
+	return x.Mul(x, new(big.Int).SetUint64(gasUsed))
+}
+
+// TestWithdrawals checks that a deposit and then withdrawal execution succeeds. It verifies the
+// balance changes on L1 and L2 and has to include gas fees in the balance checks.
+// It does not check that the withdrawal can be executed prior to the end of the finality period.
+func TestWithdrawals(t *testing.T) {
```

Comment:
Nice test, long function though. With the monorepo change the e2e tests get their own go module (so we can separate any testing dependencies, and pull in all the other modules), maybe after that we can split the `system_test.go` in themed files (general, deposits, withdrawals, p2p, etc.) or maybe even packages.

Reply:
Yea, there's a lot of boilerplate that's hard to avoid.